scripts/import/import_EPMC.pl

```diff
@@ -993,9 +1005,253 @@ sub get_current_citations{
     return \%citations;
 }
 
+###
+# Get citations from phenotype tables
+
+sub process_phenotype_feature {
+  my $dba = shift;
+  my $var_ad = shift;
+  my $pub_ad = shift;
+  my $source_ad = shift;
+  my $citation_attribs = shift;
+
+  ## Get studies from phenotype_feature
+  my $attrib_ext_sth = $dba->dbc()->prepare(qq[ select s.study_id, s.source_id, s.external_reference, s.study_type
+                                                from study s 
+                                                inner join phenotype_feature p on s.study_id = p.study_id
+                                                where p.type = 'variation' and p.study_id is not null and s.external_reference is not null
+                                                group by s.study_id, s.source_id, s.external_reference, s.study_type ]);
+  $attrib_ext_sth->execute()||die;
+  my $data = $attrib_ext_sth->fetchall_arrayref();
+
+  my $rsid_sth = $dba->dbc()->prepare(qq[ select object_id
+                                          from phenotype_feature 
+                                          where study_id = ? ]);
+
+  foreach my $l (@{$data}){
+    my $study_id = $l->[0];
+    my $source_id = $l->[1];
+    my $external_reference = $l->[2];
+    $external_reference =~ s/PMID://;
+    my $study_type = $l->[3];
+
+    # Get publication with same PMID from study table
+    my $publication = $pub_ad->fetch_by_pmid($external_reference);
+
+    next unless !defined($publication);
+
+    # Get attrib id for source - some are null 
+    my $source_attrib_id;
+    if(defined $study_type){
+      $source_attrib_id = $citation_attribs->{$study_type};
+      die "No attrib of type 'citation_source' was found for '$study_type'!\n" unless defined $source_attrib_id;
+    }
+    else{
+      # Get source name from source table (dbGaP)
+      my $source_obj = $source_ad->fetch_by_dbID($source_id);
+      my $source_name = $source_obj->name();
+      $source_attrib_id = $citation_attribs->{$source_name};
+      die "No attrib of type 'citation_source' was found for '$source_name'!\n" unless defined $source_attrib_id;
+    }
+
+    # Get publication that is not in publication table 
+    my $pub_data = get_epmc_data("search?query=ext_id:$external_reference%20src:med");
+
+    if(defined $pub_data->{resultList}->{result}->{title}){
```

Comment:
There is a lot of duplicated functionality between process_phenotype_feature_attrib & process_phenotype_feature - can you pull this out into a separate subroutine for easier maintainance?